presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraTokenSplitManager.java

```diff
@@ -13,131 +13,192 @@
+                splits.add(createSplit(subRange, endpoints));
+            }
         }
+        shuffle(splits, ThreadLocalRandom.current());
+        return unmodifiableList(splits);
+    }
 
-        // wait until we have all the results back
-        for (Future<List<TokenSplit>> futureInputSplits : splitFutures) {
-            try {
-                splits.addAll(futureInputSplits.get());
+    private Set<TokenRange> getTokenRanges()
+    {
+        return session.executeWithSession(session -> session.getCluster().getMetadata().getTokenRanges());
+    }
+
+    private Set<TokenRange> unwrap(Set<TokenRange> tokenRanges)
+    {
+        ImmutableSet.Builder<TokenRange> result = ImmutableSet.builder();
+        for (TokenRange range : tokenRanges) {
+            if (range.isWrappedAround()) {
+                result.addAll(range.unwrap());
             }
-            catch (Exception e) {
-                throw new IOException("Could not get input splits", e);
+            else {
+                result.add(range);
             }
         }
+        return result.build();
+    }
 
-        checkState(!splits.isEmpty(), "No splits created");
-        //noinspection SharedThreadLocalRandom
-        Collections.shuffle(splits, ThreadLocalRandom.current());
-        return splits;
+    private long getTotalPartitionsCount(String keyspace, String table)
+    {
+        List<SizeEstimate> estimates = getSizeEstimates(keyspace, table);
+        return estimates.stream()
+                .mapToLong(SizeEstimate::getPartitionsCount)
+                .sum();
     }
 
-    /**
-     * Gets a token range and splits it up according to the suggested
-     * size into input splits that Hadoop can use.
-     */
-    private class SplitCallable<T extends Token>
-            implements Callable<List<TokenSplit>>
+    private List<SizeEstimate> getSizeEstimates(String keyspaceName, String tableName)
     {
-        private final TokenRange range;
-        private final String keyspace;
-        private final String columnFamily;
-        private final int splitSize;
-        private final CassandraThriftClient client;
-        private final IPartitioner partitioner;
-
-        public SplitCallable(TokenRange range, String keyspace, String columnFamily, int splitSize, CassandraThriftClient client, IPartitioner partitioner)
-        {
-            checkArgument(range.rpc_endpoints.size() == range.endpoints.size(), "rpc_endpoints size must match endpoints size");
-            this.range = range;
-            this.keyspace = keyspace;
-            this.columnFamily = columnFamily;
-            this.splitSize = splitSize;
-            this.client = client;
-            this.partitioner = partitioner;
+        checkSizeEstimatesTableExist();
+
+        Statement statement = select("range_start", "range_end", "mean_partition_size", "partitions_count")
+                .from(SYSTEM, SIZE_ESTIMATES)
```

Comment:
No. It is not. If authorization is enabled, DBA must grant access to this table to the user which is being used in cassandra config.